core/src/main/java/hudson/Util.java

```diff
@@ -1688,6 +1689,52 @@ public static void closeAndLogFailures(@CheckForNull Closeable toClose, @Nonnull
         }
     }
 
+    @Restricted(NoExternalUse.class)
+    public static int permissionsToMode(Set<PosixFilePermission> permissions) {
```

Comment:
Surprises me that `PosixFilePermissions` does not include this obvious-seeming function.